### DISCUSSION TITLE
Posible resolución para la parte C del parcial de QMP. Está bien?

### DISCUSSION THREAD

**Discussion by julchat** [dds-jv/foro#159] [Consultas de UI / REST] [answered]
Hola, con un amigo estuvimos haciendo uno de los parciales, y tuvimos dudas sobre si estabamos haciendo las cosas bien en la parte de presentación (C). Esta fue la solución que llegamos

![QMP Parcial 2 web](https://user-images.githubusercontent.com/48997230/100042356-09815480-2dea-11eb-815d-f7c822d27058.png)

 El metodo para GET /Usuarios/:id/Prendas?pagina= en el controller deberá ir a buscar a la base de datos todas las prendas que existan y cargarlas en memoria, allí va a dividirlas por páginas (se hace de esta forma para que la navegación por página sea fluida y no haya que hacer una query cada vez). Si no se especifica pagina o se recibe ?pagina= se mostrará las prendas que pertenezcan a la primera página. Esa información es la que va en el viewModel en forma de Cards.

Por otro lado el método para GET /Usuarios/:id/Prendas/new deberá ir a buscar para proporcionar las opciones del Select las categorías persistidas.

Por último, POST /Usuarios/:id/Prendas/new deberá persistir en nuestra base de datos, a partir del form data en el body, una nueva prenda para dicho usuario. Posteriormente, nos redirigirá al guardarropas del usuario. (/Usuarios/:id/Prendas)
 
Está bien eso? Si le pueden echar un ojo estaría buenísimo.
Gracias y saludos,
Cristian

**Comment by gastonprieto** [comment]
Aloha!

Siempre el link es bienvenido :innocent: 

La primer pantalla la veo ok, tal vez no pondría el carousel, mas que nada porque hay que explicar como se hace, ya que no es nativo.

La 2da la veo ok.

> El metodo para GET /Usuarios/:id/Prendas?pagina= 

En general en la ruta pones el usuario si es que por ejemplo podrías ver prendas de múltiples usuarios, si las prendas que ves son la del usuario logueado en general se suele omitir.

> en el controller deberá ir a buscar a la base de datos todas las prendas que existan y cargarlas en memoria, allí va a dividirlas por páginas (se hace de esta forma para que la navegación por página sea fluida y no haya que hacer una query cada vez). 

Esto suena raro explicado, si vos dividis por página, que renderizas una única página en la pantalla? Si mostras una página entonces vas a ir varias veces a la base de datos, porque cada vez que cambien de página tienen que hacer una request nueva para pedir la información de la nueva página. Sino tenes que explicar como haces desde el cliente para paginar. Mucho ojo con eso, porque si haces cosas que no vimos o mas complejas deberías explicar como lo vas a hacer al detalle.

> Si no se especifica pagina o se recibe ?pagina= se mostrará las prendas que pertenezcan a la primera página. Esa información es la que va en el viewModel en forma de Cards.

Esto me refuerza la idea de que igual vas a tener n requests.

> Por otro lado el método para GET /Usuarios/:id/Prendas/new deberá ir a buscar para proporcionar las opciones del Select las categorías persistidas.

Joya

> Por último, POST /Usuarios/:id/Prendas/new deberá persistir en nuestra base de datos, a partir del form data en el body, una nueva prenda para dicho usuario. Posteriormente, nos redirigirá al guardarropas del usuario. (/Usuarios/:id/Prendas)

Ok, pero la ruta esta mal, eso indica que estas tratando de crear o reemplazar un formulario (recurso que te dio el GET) vos lo que queres es crear una nueva prenda y para eso es un POST a .../prendas

**Comment by julchat** [comment]
Todo claro! Muchas gracias!

**Comment by julchat** [comment, marked as answer]
Un poco tarde para el link, pero lo dejo por si alguno entra y quiere tenerlo a mano
https://docs.google.com/document/d/15ifQWQqfzbwwHIiiuoZiE4eyhd_DpmaQ5OGPvD9x2Wc/edit#heading=h.uyku9mnteh0t
